### PR TITLE
updates for 0.7

### PR DESCRIPTION
**Pull request by stkb**
Summary of changes ([purescript 0.7 migration guide](https://github.com/purescript/purescript/wiki/0.7-Migration-Guide))
### Dependencies
* All bower dependencies updated for 0.7 compatibility
* Express updated: 4.11 -> 4.13

### Language
* No more implicit prelude import: `import Prelude` added to files
* `[a]` syntax for array types replaced with `Array a`
* `Debug.Trace.trace` replaced with `Control.Monad.Eff.Console.log`
* Number type replaced with Int where appropriate (status code)
* Inline FFI javascript definitions no longer allowed: moved to separate .js files (with same name as .purs file)
* Infix operators in instance definitions replaced with function names
* Cons pattern matching (x : xs) removed; replaced with folds or other functions (`QueryString.purs`)
* Also in `QueryString.purs`, Text.Parsing.Parser api was modified slightly

### Build system
Biggest change was to the build system. grunt-purescript no longer works with purescript 0.7 (and looks unlikely that it will be updated soon). I haven't taken a dependency on another build tool but have instead written build scripts in `package.json` that use the native purescript compiler tools:
* `npm install`: install all bower & npm dependencies and build the project
* `npm run build`: builds package & creates docs
* `npm test`: builds, creates docs and builds & runs tests
* `npm run build-examples`: builds package and all examples (examples are output to "./output/examples/" I also had to modify the namespaces in the examples to make this work)

Some warnings come up during build, these are from dependencies and don't do any harm.

These build scripts work fine, but using [pulp](https://github.com/bodil/pulp) for building might be something to consider (a lot of other purescript packages have moved to pulp). Unline grunt & gulp, pulp requires no extra dependencies to be installed with the package nor config files written; it just needs to be installed as a global npm package on your machine.

### CODE REVIEW

**Comment by stkb** [comment]
I forgot about travis. Thought i'd fixed it but the travis build's still failing because it's looking for grunt (which is no longer there)

File: src/Node/Express/Types.purs [hunk 1]
```diff
@@ -78,4 +79,4 @@ newtype CookieOptions = CookieOptions { maxAge :: Number, signed :: Boolean, pat
 
 instance defaultCookieOptions :: Default CookieOptions where
     def = CookieOptions { maxAge: oneYear, signed: false, path: "/" }
-      where oneYear = 365 * 24 * 60 * 60 * 1000
+      where oneYear = 365.0 * 24.0 * 60.0 * 60.0 * 1000.0
```
Comment (michaelficarra) [inline src/Node/Express/Types.purs]: Wouldn't it be better to change CookieOptions.maxAge to an Int?

**Comment by michaelficarra** [comment]
To fix Travis, you need to update [`.travis.yml`](https://github.com/dancingrobot84/purescript-express/blob/master/.travis.yml) with new build and test commands.

**Comment by stkb** [comment]
Ok thanks, I've made the changes to maxAge and to the travis build

**Comment by michaelficarra** [comment]
This looks great. :shipit:

**Comment by michaelficarra** [comment]
@stkb I tried this out in my 0.7 project and everything worked flawlessly. Upgrade was painless.

@dancingrobot84 I think this is good to go.

**Comment by stkb** [comment]
@michaelficarra Good to hear; thanks for the endorsement!

File: bower.json [hunk 2]
```diff
@@ -20,20 +20,22 @@
...
+    "purescript-either": "0.2.x",
+    "purescript-strings": "0.5.x",
+    "purescript-easy-ffi": "purescript-0.7-prerelease",
+    "purescript-control": "0.3.x",
+    "purescript-parsing": "0.4.x",
+    "purescript-foldable-traversable": "0.4.x",
+    "purescript-exceptions": "0.3.x",
+    "purescript-eff": "0.1.0",
+    "purescript-lists": "0.7.x",
+    "purescript-console": "0.1.x"
```
Comment (dancingrobot84) [inline bower.json]: Could you please move this one to `devDependencies`? AFAIK it is used only in examples.

File: examples/EmbedApp.purs [hunk 3]
```diff
@@ -19,19 +20,7 @@ attach = apply appSetup
 -- Pretend this is some external JS file and we want
 -- to attach our PS handlers into express app that
 -- already exists there.
```
Comment (dancingrobot84) [inline examples/EmbedApp.purs]: I guess this comment could be removed too since we do not need to pretend no more:)

**Comment by dancingrobot84** [comment]
This one is a nice work. Thank you very much! As soon as some minor flaws are fixed I will accept your PR.

**Comment by stkb** [comment]
Ok, have
- moved purescript-console to devDependencies
- removed comment in EmbedApp.purs

and squashed the commits.

**Comment by dancingrobot84** [comment]
Great! Thank you.

**Comment by stkb** [comment]
Thanks!

**Pull request merged**